.github/workflows/tests@v1.yml

```diff
@@ -28,9 +28,33 @@ jobs:
           java-version: ${{ matrix.java-version }}
           distribution: 'adopt'
 
+      - name: Get POM hash
+        id: get-pom-hash
+        run: echo "value=${{ hashFiles('**/pom.xml') }}" >> "$GITHUB_OUTPUT"
+
+      - uses: actions/cache/restore@v4
+        id: java-cache
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-${{ matrix.java-version }}-maven-${{ steps.get-pom-hash.outputs.value }}
+
       - name: Compile source and tests
         run: mvn -B compile test-compile -Dfmt.skip=true -Dclirr.skip=true -Danimal.sniffer.skip=true
 
+      - name: Download test dependencies
+        if: steps.java-cache.outputs.cache-hit != 'true'
+        run: mvn -B test -Dtest=TestThatDoesNotExists || true
```

Comment:
I think this will cause `test` to fail on `core` module due to no tests being executed and not go further to other modules

Reply:
During download step it also may be worth to have those skips `-Dfmt.skip=true -Dclirr.skip=true -Danimal.sniffer.skip=true`

Reply:
> I think this will cause `test` to fail on `core` module due to no tests being executed and not go further to other modules

It works as expected.

Reply:
> During download step it also may be worth to have those skips `-Dfmt.skip=true -Dclirr.skip=true -Danimal.sniffer.skip=true`

Added